### PR TITLE
Add factory_bot gem

### PR DESCRIPTION
**Pull request by gdiasdasilva**
This pull request will add the `factory_bot_rails` gem to the project, configure the main settings for its usage in tests and replace `dev:prime` creation of users to take advantage of the newly added helpers.

### CODE REVIEW

File: lib/tasks/dev.rake [hunk 1]
```diff
@@ -2,12 +2,9 @@ if Rails.env.development? || Rails.env.staging?
...
     task prime: "db:setup" do
-      User.create(
-        first_name: "John",
-        last_name: "Wayne",
-        email: "john@example.org",
-        password: "stagecoach"
-      )
+      include FactoryBot::Syntax::Methods
+
+      create(:user, first_name: "John", last_name: "Wayne")
```
Comment (ruiccarvalho) [inline lib/tasks/dev.rake:7]: Not really a fan of using factories outside of tests.

The point of factories is to use "inheritance" and "composition" to ease setup data where we only want some attributes for testing purposes but we still need to fill the rest to pass the validations.

In this task, we want real-world-like data, where all attributes matter, created with the same ActiveRecord statements the controllers, models, and services are using.

It also creates an unnecessary coupling with an external dependency.

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -19,8 +19,7 @@
 # of increasing the boot-up time by auto-requiring all files in the support
 # directory. Alternatively, in the individual `*_spec.rb` files, manually
 # require only the support files necessary.
-#
-# Dir[Rails.root.join('spec', 'support', '**', '*.rb')].each { |f| require f }
+Dir[Rails.root.join('spec', 'support', '**', '*.rb')].each { |f| require f }
```
Comment (ruiccarvalho) [inline spec/rails_helper.rb:22]: I think this line won't be necessary when the code from #14 and its cleanup of the rspec configuration enters `master`.

**Review by ruiccarvalho — CHANGES_REQUESTED** [review]
One point for debate. Anyway, this should wait for #11 and #14 (merged in this order) and then be rebased with `master` to re-check.

File: lib/tasks/dev.rake (hunk 1 above)
Comment (gdiasdasilva) [reply lib/tasks/dev.rake:7]: I agree, although it doesn't seem like a blocking issue. It was following thoughtbot's examples in their [suspenders](https://github.com/thoughtbot/suspenders/blob/master/templates/dev.rake) repo.

File: spec/rails_helper.rb (hunk 2 above)
Comment (gdiasdasilva) [reply spec/rails_helper.rb:22]: 👍

**Pull request closed without merging**